### PR TITLE
ree-1.8.7 requires old version of OpenSSL 1.0.2

### LINKED ISSUE #4110: ree-1.8.7 requires old version of OpenSSL 1.0.2
## Description

I am attempting to reinstall ree-1.8.7-2012.02. The install appears to work against the latest openssl available from brew which is openssl@1.1 

However the gem command fails with a missing symbol.

I then gave up on trying to use the latest version of openssl and tried to get back to a known configuration.

I know that the Ruby build will work against openssl 1.0.2j and I managed to get that installed (compiled by hand from the old openssl 1.0.2j tarball) and installed in homebrew successfully.

I sent rvm autolibs read in an attempt to get it not to try to use the latest openssl but the requirements still processes as openssl@1.1 and it downloads the latest and installs against that.

## Steps to reproduce

ENVIRONMENT

Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.G8uhq73TcT/Render
BASH=/bin/bash
BASH_ARGC=()
BASH_ARGV=()
BASH_COMMAND='set > t'
BASH_LINENO=()
BASH_REMATCH=()
BASH_SOURCE=()
BASH_SUBSHELL=0
BASH_VERSINFO=([0]="3" [1]="2" [2]="57" [3]="1" [4]="release" [5]="x86_64-apple-darwin15")
BASH_VERSION='3.2.57(1)-release'
COLORFGBG='15;0'
COLUMNS=130
COMMAND_MODE=unix2003
DIRSTACK=()
EUID=1616278830
GEM_HOME=/Users/gamblew/.rvm/gems/ruby-1.8.7-head
GEM_PATH=/Users/gamblew/.rvm/gems/ruby-1.8.7-head:/Users/gamblew/.rvm/gems/ruby-1.8.7-head@global
GROUPS=()
HISTFILE=/Users/gamblew/.bash_history
HISTFILESIZE=500
HISTSIZE=500
HOME=/Users/gamblew
HOSTNAME=C02M43AZFD59
HOSTTYPE=x86_64
IFS=$' \t\n'
IRBRC=/Users/gamblew/.rvm/rubies/ruby-1.8.7-head/.irbrc
ITERM_ORIG_PS1='\W \! \[\033[0;31m\]$(parse_git_branch) \[\033[30;32m\]($(parse_rvm_info))\[\e[0m\]\n$ '
ITERM_PREEXEC_INTERACTIVE_MODE=
ITERM_PREV_PS1=$'\\[\E]133;D;$?\a\E]1337;RemoteHost=gamblew@C02M43AZFD59\a\E]1337;CurrentDir=/Users/gamblew\a\E]133;A\a\\]\\W \\! \\[\\033[0;31m\\]$(parse_git_branch) \\[\\033[30;32m\\]($(parse_rvm_info))\\[\\e[0m\\]\\n$ \\[\E]133;B\a\\]'
ITERM_PROFILE=Default
ITERM_SESSION_ID=w0t0p0:CD4F6C16-2DF5-480B-A7E7-0AC1911B498A
ITERM_SHELL_INTEGRATION_INSTALLED=Yes
JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_65.jdk/Contents/Home
LANG=en_US.UTF-8
LINES=39
LOGNAME=gamblew
MACHTYPE=x86_64-apple-darwin15
MAILCHECK=60
MY_RUBY_HOME=/Users/gamblew/.rvm/rubies/ruby-1.8.7-head
OLDPWD=/Users/gamblew/.rvm/src/ree-1.8.7-2012.02
OPTERR=1
OPTIND=1
OSTYPE=darwin15
PATH=/Users/gamblew/.rvm/gems/ruby-1.8.7-head/bin:/Users/gamblew/.rvm/gems/ruby-1.8.7-head@global/bin:/Users/gamblew/.rvm/rubies/ruby-1.8.7-head/bin:/usr/local/opt/openssl/bin:/usr/local/bin:/usr/local/sbin:.:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/gamblew/.rvm/bin
PIPESTATUS=([0]="0")
PKG_CONFIG_PATH=/usr/local/opt/openssl@1.1/lib/pkgconfig
PPID=1512
PROMPT_COMMAND=iterm2_preexec_invoke_cmd
PS1='\W \! \[\033[0;31m\]$(parse_git_branch) \[\033[30;32m\]($(parse_rvm_info))\[\e[0m\]\n$ '
PS2='> '
PS4='+ '
PWD=/Users/gamblew
RUBY_VERSION=ruby-1.8.7-head
SECURITYSESSIONID=186a7
SHELL=/bin/bash
SHELLOPTS=braceexpand:functrace:hashall:histexpand:history:interactive-comments:monitor:vi
SHLVL=1
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.epHVTlIKSg/Listeners
SVN_EDITOR=vi
TERM=xterm-256color
TERM_PROGRAM=iTerm.app
TERM_PROGRAM_VERSION=3.0.15
TERM_SESSION_ID=w0t0p0:CD4F6C16-2DF5-480B-A7E7-0AC1911B498A
TITLEBAR='\[\033]0;\u@\h:\w\007\]'
TMPDIR=/var/folders/w1/b7_bp23n40sdvrb7ts3hm5xnh5cx9f/T/
UID=1616278830
USER=gamblew
XPC_FLAGS=0x0
XPC_SERVICE_NAME=0
_=ree-1.8.7-2012.02
__CF_USER_TEXT_ENCODING=0x6056752E:0x0:0x0
_system_arch=x86_64
_system_info='Darwin C02M43AZFD59 15.6.0 Darwin Kernel Version 15.6.0: Sun Jun  4 21:43:07 PDT 2017; root:xnu-3248.70.3~1/RELEASE_X86_64 x86_64'
_system_name=OSX
_system_name_lowercase=osx
_system_type=Darwin
_system_version=10.11
chpwd_functions=([0]="__rvm_cd_functions_set")
entry=list
expanded_alias_name=
hist_ent='  546  set > t'
iterm2_hostname=C02M43AZFD59
iterm2_ran_preexec=yes
last_hist_ent='  546  set > t'
old_rvm_debug_flag=0
old_rvm_verbose_flag=0
reinstall_flag=1
rvm_autoupdate_flag=2
rvm_bin_path=/Users/gamblew/.rvm/bin
rvm_configure_flags=([0]="--with-openssl-dir=/usr/local/Cellar/openssl/1.0.2j")
rvm_ignore_rvmrc=0
rvm_path=/Users/gamblew/.rvm
rvm_prefix=/Users/gamblew
rvm_pretty_print_flag=auto
rvm_ruby_string=ree-1.8.7-2012.02
rvm_rubygems_path=/Users/gamblew/.rvm/rubygems
rvm_saved_env=()
rvm_version='1.29.2 (master)'
rvmrc=/Users/gamblew/.rvmrc
script=override_gem
string=2012.02
strings=([0]="ree" [1]="1.8.7" [2]="2012.02")

## Expected behavior

I expected that the old openssl would be seen as sufficient for the openssl requirement and be used (note the CPPFLAG, LDFLAGS etc. in the gist)

## Actual behavior

https://gist.github.com/weyus/460e5a50e4b0944914945e7e82ef2010

## Environment info

$ rvm info

ruby-1.8.7-head:

  system:
    uname:        "Darwin C02M43AZFD59 15.6.0 Darwin Kernel Version 15.6.0: Sun Jun  4 21:43:07 PDT 2017; root:xnu-3248.70.3~1/RELEASE_X86_64 x86_64"
    name:         "OSX"
    version:      "10.11"
    architecture: "x86_64"
    bash:         "/bin/bash => GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin15)"
    zsh:          "/bin/zsh => zsh 5.0.8 (x86_64-apple-darwin15.0)"
    remote_path:  "osx/10.11/x86_64"

  rvm:
    version:      "1.29.2 (master)"
    updated:      "21 hours 51 minutes 59 seconds ago"
    path:         "/Users/gamblew/.rvm"
    autolibs:     "[read] Allow RVM to use package manager if found but do not install or fail if dependencies are missing."

  ruby:
    interpreter:  "ruby"
    version:      "1.8.7"
    date:         "2014-01-28"
    platform:     "i686-darwin13.3.0"
    patchlevel:   "2014-01-28 patchlevel 376"
    full_version: "ruby 1.8.7 (2014-01-28 patchlevel 376) [i686-darwin13.3.0]"

  homes:
    gem:          "/Users/gamblew/.rvm/gems/ruby-1.8.7-head"
    ruby:         "/Users/gamblew/.rvm/rubies/ruby-1.8.7-head"

  binaries:
    ruby:         "/Users/gamblew/.rvm/rubies/ruby-1.8.7-head/bin/ruby"
    irb:          "/Users/gamblew/.rvm/rubies/ruby-1.8.7-head/bin/irb"
    gem:          "/Users/gamblew/.rvm/rubies/ruby-1.8.7-head/bin/gem"
    rake:         "/Users/gamblew/.rvm/gems/ruby-1.8.7-head/bin/rake"

  environment:
    PATH:         "/Users/gamblew/.rvm/gems/ruby-1.8.7-head/bin:/Users/gamblew/.rvm/gems/ruby-1.8.7-head@global/bin:/Users/gamblew/.rvm/rubies/ruby-1.8.7-head/bin:/usr/local/opt/openssl/bin:/usr/local/bin:/usr/local/sbin:.:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/gamblew/.rvm/bin"
    GEM_HOME:     "/Users/gamblew/.rvm/gems/ruby-1.8.7-head"
    GEM_PATH:     "/Users/gamblew/.rvm/gems/ruby-1.8.7-head:/Users/gamblew/.rvm/gems/ruby-1.8.7-head@global"
    MY_RUBY_HOME: "/Users/gamblew/.rvm/rubies/ruby-1.8.7-head"
    IRBRC:        "/Users/gamblew/.rvm/rubies/ruby-1.8.7-head/.irbrc"
    RUBYOPT:      ""
    gemset:       ""

### PR DESCRIPTION
**Pull request by pkuczynski**
Fixes #4110

### CODE REVIEW

File: scripts/functions/requirements/osx_brew [hunk 1]
```diff
@@ -315,6 +315,10 @@ requirements_osx_brew_libs_default()
 
   # OpenSSL version depends on the version of Ruby being installed
   case "$1" in
+    (ree-1.8*)
+      brew_openssl_package="openssl"
+      ;;
+
     (ruby-2.3*|ruby-2.2*|ruby-2.1*|ruby-2.0*|ruby-1.9*|ruby-1.8*)
```
Comment (mpapis) [inline scripts/functions/requirements/osx_brew]: this could be: `(ruby-2.3*|ruby-2.2*|ruby-2.1*|ruby-2.0*|ruby-1.*|ree*)`?

Comment (pkuczynski) [reply scripts/functions/requirements/osx_brew]: Yes, it could. But I preferred to keep it separate for clarity. Don't you think its better?

Comment (mpapis) [reply scripts/functions/requirements/osx_brew]: both work :) our code grows huge so I thought a bit shorter would not harm, maybe on beginning? but go ahead and merge if you like

Comment (pkuczynski) [reply scripts/functions/requirements/osx_brew]: Will get smaller once we re-write to ruby :)

**Pull request merged**